language/golang/language.go

```diff
@@ -106,8 +104,33 @@ func (l *Language) Execute(logger *log.Logger, repositoryPath string) (coverage
 		Directory: repositoryPath,
 	})
 	if err != nil {
-		return 0, pkgerrors.WithMessage(pkgerrors.WithStack(err), commandOutput)
+		if strings.Contains(commandOutput, "ERROR: no test files found") {
+			return 0, problems, pkgerrors.WithMessage(pkgerrors.WithStack(err), commandOutput)
+		}
+
+		testSummary := languageGoTestsErrorMatch.FindStringSubmatch(commandOutput)
+		if len(testSummary) > 0 {
```

Comment:
I think if we cannot find the test summary it should be a hard error.

Reply:
No. We are explicitly looking for "DONE x tests, x error". If there are no errors, then it's "DONE x tests" only. And we are only interested in the case where there are errors.

Reply:
Also, we are just looking in the error case. And in the error case we cannot rely on the summary being there.

Reply:
> If there are no errors, then it's "DONE x tests" only.

So then there are no errors, but... we have `err != nil`. Sound like a hard error to me.

Reply:
If there is a failing test then it is "DONE 1 tests, 1 failure", but we still have `err != nil` because the exist status is `1`.

Reply:
Right. And you said if our regex does not find anything, we might have "Done 1 tests" which means no error? But we are still in the block `err != nil` so... something must be off? So hard error?

Reply:
I think we went straight into confusion. Let me try again.

We have these scenarios:

- DONE 1 tests
- DONE 0 tests, 1 error -> exit status 1
- DONE 1 tests, 1 failure -> exit status 1

That is, in the first case we have `err == nil`, so we don't need the regex.
For the other two cases we have `err != nil`, but only the first one is a hard error since this indicates a syntax error. For the last case it's just a soft error since we still can get a coverage.

Reply:
We can turn the logic to just look for "DONE 1 tests, 1 failure", and everything else is a hard error.

Reply:
> We can turn the logic to just look for "DONE 1 tests, 1 failure", and everything else is a hard error.

did that.